pkg/vault/contracts/VaultAdmin.sol

```diff
@@ -270,28 +270,33 @@ contract VaultAdmin is IVaultAdmin, VaultCommon, Authentication {
     }
 
     /// @inheritdoc IVaultAdmin
-    function collectAggregateFees(address pool) public onlyVaultDelegateCall nonReentrant withRegisteredPool(pool) {
+    function collectAggregateFees(
+        address pool
+    )
+        public
+        onlyVaultDelegateCall
+        onlyWhenUnlocked
+        onlyProtocolFeeController
+        withRegisteredPool(pool)
```

Comment:
Note for the reviewers: modifiers have changed.

- There are no external calls here, so `nonReentrant` was removed. I've also removed `receiveAggregateFees` from this function, so it could be re-added if you think there's a reason I might be missing.
- `onlyWhenUnlocked` is now a must because this function supplies credit. Not having this modifier is a critical issue; there's a test for that too.
- `onlyProtocolFeeController` is now a must because otherwise anyone can claim the credit granted by calling this function, and then steal the fees. There's also a test for that.

Reply:
Was there a reason we used `withRegisteredPool` instead of `withInitializedPool` here? There wouldn't be fees on a pool until it's initialized, right? (The tests would have to be updated to initialize the pool before calling this.)

Reply:
I think it doesn't make much of a difference; a pool that is not initialized will have no fees and a pool that was just initialized pool won't have fees either.